ee/server/service/software_installers.go

```diff
@@ -304,6 +304,21 @@ func (svc *Service) UpdateSoftwareInstaller(ctx context.Context, payload *fleet.
 	}
 	payload.ValidatedLabels = validatedLabels
 
+	catIDs, err := svc.ds.GetSoftwareCategoryIDs(ctx, payload.Categories)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "getting software category ids")
+	}
+
+	if len(catIDs) != len(payload.Categories) {
+		return nil, &fleet.BadRequestError{
+			Message:     "some or all of the categories provided don't exist",
+			InternalErr: fmt.Errorf("categories provided: %v", payload.Categories),
+		}
+	}
+
+	payload.CategoryIDs = catIDs
+	dirty["Categories"] = true
```

Comment:
We'll need to revise dirty checking such that categories changes don't trigger side effects (cancelling pending actions), same as turning self-service on/off, as this action is more minor than that one.

We have a special function for setting the self-service flag (without side effects) right now that's only used here, so that function can be expanded to handle categories as well without breaking anything else.

/cc @RachelElysia because if we're showing the "cxl pending" warning when only categories were edited, we shouldn't, as of this tweak.

Reply:
If it's cleaner to drop the special function entirely and do something more complex than passing `true` to "was metadata changed" on the side effects function, that works for me.